rdoxygen-design.md

```diff
@@ -0,0 +1,126 @@
+                      )
+
+# add Doxyfile in package
+doxy_init <- function(pkg = ".", # any subfolder of package root
+                      doxyfile = "inst/doc/doxygen/Doxyfile", # path to doxyfile relative to package root
+                      options # passed to doxy_edit
+                      )
+
+# wrap Doxygen documentation in R vignette
+doxy_vignette <- function(pkg = ".", # any subfolder of package root
+                          doxyfile = "inst/doc/doxygen/Doxyfile", # path to doxyfile relative to package root
+                          vignetteName = "DoxygenVignette.Rmd", # name of Doxygen vignette
+                          indexEntry # name of vignette Index Entry.  defaults to "C++ library documentation for package PackageName"
+                          )
+
+# do all steps above simultaneously
+doxy <- function(pkg = ".", # any subfolder of package root
+                 doxyfile = "inst/doc/doxygen/Doxyfile", # path to doxyfile relative to package root
+                 options, # passed to doxy_edit
+                 vignette = TRUE # add vignette
+                 vignetteName = "DoxygenVignette.Rmd", # passed to doxy_vignette
+                 indexEntry # passed to doxy_vignette
+                 )
+```
+
+## Default File Locations
+
+* *Doxygen documentation.*  Installed R vignettes can only display HTML files stored in a subfolder of `inst/doc` (as documented [here](https://github.com/nevrome/rdoxygen/issues/2#issuecomment-412536748)).  Therefore, the suggested location for Doxygen documentation is `inst/doc/doxygen`.
+
+* *Doxyfile.*  As a package developer, I feel like whatever is required to create the package exactly as it should be installed on disk should be part of the package itself.  In this sense, the `Doxyfile` needed to format the Doxygen documentation exactly as I want it should be part of the package as well.  
+
+    An obvious location for this file is `inst/doc/doxygen`.  However, `inst/doc` is typically `.gitignore`d (e.g., by `devtools::use_vignette()` and `usethis::use_vignettes()`).  So we'd have to manually exclude `inst/doc/doxygen/Doxyfile`, via appending an existing `.gitignore` with something like [this](https://stackoverflow.com/questions/5533050/gitignore-exclude-folder-but-include-specific-subfolder):
+    
+    ```
+    # paste the following to the bottom of existing .gitignore
+    
+    !inst/doc # unignores inst/doc
+    inst/doc/* # ignore everything inside inst/doc but not inst/doc itself
+    !inst/doc/doxygen/Doxyfile # unignore Doxyfile
+    ```
+    
+    Another option is to have `Doxyfile` live in `vignettes`.  If creating Doxgygen vignettes is always enabled (i.e., not optional), then this folder already exists, so easier to put `Doxyfile` here than to edit the `.gitignore` as above.  However, if the folder doesn't exist, then perhaps it's confusing to create the folder just to store the `Doxyfile`.
```

Comment:
Yes - I prefer the git unignore solution

Reply:
Me too :)  I tested this feature a bit and it's somewhat delicate (seems like some versions of git only unignore one level deep).  Since we just want to unignore the `Doxyfile` it shouldn't be too hard...